charts/prometheus-ipmi-exporter/templates/deployment.yaml

```
@@ -56,8 +56,16 @@ spec:
               memory: {{ .Values.resources.requests.memory }}
       volumes:
         - name: config-volume
+        {{- if .Values.useSecret }}
+          secret:
+            secretName: {{ .Values.secret.name }}
+            items:
+            - key: {{ .Values.secret.key }}
+              path: config.yml
+        {{- else }}
```

With respect to the comment above:
```suggestion
        {{- if .Values.configSecret.enabled }}
          secret:
            secretName: {{ .Values.configSecret.name }}
            items:
            - key: {{ .Values.configSecret.key }}
              path: config.yml
        {{- else }}
```